ccmlib/node.py

```diff
@@ -560,19 +576,32 @@ def watch_log_for(self, exprs, from_mark=None, timeout=600, process=None, verbos
                             if len(tofind) == 0:
                                 return matchings[0] if isinstance(exprs, string_types) else matchings
                 else:
-                    # yep, it's ugly
+                    # wait for the situation to clarify, either stop or just a pause in log production
                     time.sleep(1)
-                    if start + timeout < time.time():
-                        raise TimeoutError(time.strftime("%d %b %Y %H:%M:%S", time.gmtime()) + " [" + self.name + "] Missing: " + str([e.pattern for e in tofind]) + ":\n" + reads[:50] + ".....\nSee {} for remainder".format(filename))
 
-                if process:
-                    if common.is_win():
-                        if not self.is_running():
-                            return None
-                    else:
-                        process.poll()
-                        if process.returncode == 0:
-                            return None
+                    # check if abort condition is not met e.g. C* process is terminated
+                    if abort_function is not None:
```

Comment:
why do we call it here?

Reply:
ok, the logic of abort function mixed me up - by calling `abort_function` I conclude that we want to abort at this point. But it is more like maybe_abort... can be this more explicit? Maybe just pass a bool param - can_abort or something and then inline that cancellation code?

Reply:
I was thinking about it as well... but the problem is this `watch_for_logs` is generic, used in tons of places.
So I wanted to have control about the condition that is used to abort...

I can try to change it again and instead of a function use a boolean: `stop_if_cassandra_pid_terminated`...

Reply:
or just pass a condition